src/serializer/ResidualHeapSerializer.js

```diff
@@ -365,11 +365,13 @@ export class ResidualHeapSerializer {
       // conditional assignment
       values.push(cond);
       let consequent = absVal.args[1];
-      invariant(consequent instanceof AbstractValue);
+      if (consequent instanceof AbstractValue) {
```

Comment:
I don't understand why non-abstract values in the consequent/alternate positions can be ignored. Similarly, in the emitter code a bit later. Is the only allowed non-abstract value the `EmptyValue`? In that case, should a deletion statement be emitted below?

Reply:
Also, it should be possible to craft a really small serializer regression test without optimized functions that hits this case.

Reply:
Why do we need to delete in the case of an `EmptyValue`? I tried that just now but the output made absolutely no sense – there is no reason to `delete` anything here.

Reply:
@hermanventer It sounds like this PR might not be the right fix then. Can you give me some pointers on what I need to do to fix this correctly? If you'd be happier to take a look later when your looking into the fixed point logic, I'd understand.

Reply:
Coming back to the question of what to do with EmptyValue: When this value "leaks" outside of code that checks/ensures property absence, it must be "promoted" to undefined. There is a helper method "promoteEmptyToUndefined" on Value. Currently it is only used to guard against EmptyValue being written into a variable or property. It seems that it should also be used to guard against EmptyValue leaking out of an optimized function.

Reply:
The EmptyValue problem is fixed in https://github.com/facebook/prepack/pull/1787